### PR TITLE
Rangecheck pub input

### PR DESCRIPTION
**Pull request by MauroToscano**

### CODE REVIEW

**Review by diegokingston — APPROVED** [review]

**Review by gabrielbosio — APPROVED** [review]

File: src/cairo/runner/run.rs [hunk 1]
```diff
@@ -94,14 +95,28 @@ pub fn run_program(
...
+
+    Ok((register_states, cairo_mem, data_len, range))
 }
 
 pub fn generate_prover_args(
     file_path: &str,
     rc_builtin_range: Option<Range<u64>>,
 ) -> (TraceTable<Stark252PrimeField>, CairoAIR, PublicInputs) {
-    let (register_states, memory, program_size) =
+    let (register_states, memory, program_size, _rg_init_end) =
```
Comment (entropidelic) [inline src/cairo/runner/run.rs:119]: if we are getting the range of memory addresses from the cairo vm, we should not ignore the variable `rg_init_end`. 
`generate_prover_args` should not receive the hardcoded`rc_builtin_range` now. That value should be calculated with the feature added in this PR.

Comment (entropidelic) [reply src/cairo/runner/run.rs:119]: Also, same comment as before, let's be consistent with the namings

**Review by entropidelic — CHANGES_REQUESTED** [review]
good work, just some minor fixes

File: src/cairo/runner/run.rs [hunk 2]
```diff
@@ -94,14 +95,28 @@ pub fn run_program(
...
     let data_len = cairo_runner.get_program().data_len();
 
-    Ok((register_states, cairo_mem, data_len))
+    // get range start and end
+    let range = vm
+        .get_range_check_builtin()
+        .map(|builtin| {
+            let (idx, stop_offset) = builtin.get_memory_segment_addresses();
+            let stop_offset = stop_offset.unwrap_or_default();
+            let rangecheck_base =
```
[the remark this answers is not in the document]
Comment (pablodeymo) [reply src/cairo/runner/run.rs]: thanks!

File: src/cairo/runner/run.rs (hunk 1 above)
Comment (pablodeymo) [reply src/cairo/runner/run.rs]: fixed! thanks!

**Review by entropidelic — APPROVED** [review]

**Pull request merged**